plugins/modules/identity/sap_user.py

```diff
@@ -0,0 +1,510 @@
+    logondata = dict()
+    company = dict()
+    # for change parameters
+    addressx = dict()
+    passwordx = dict()
+    logondatax = dict()
+    companyx = dict()
+    force = bool()
+    # define username
+    add_to_dict(params, 'USERNAME', username)
+    # define Address
+    add_to_dict(address, 'FIRSTNAME', firstname)
+    add_to_dict(address, 'LASTNAME', lastname)
+    add_to_dict(address, 'E_MAIL', email)
+    # define Password
+    add_to_dict(password, 'BAPIPWD', raw_password)
+    # define Alias
+    add_to_dict(alias, 'USERALIAS', useralias)
+    # define LogonData
+    add_to_dict(logondata, 'GLTGV', datetime.date.today())
+    add_to_dict(logondata, 'GLTGB', '20991231')
+    add_to_dict(logondata, 'USTYP', user_type)
+    # define company
+    add_to_dict(company, 'COMPANY', raw_company)
+    #
+    params['LOGONDATA'] = logondata
+    params['ADDRESS'] = address
+    params['COMPANY'] = company
+    params['ALIAS'] = alias
+    params['PASSWORD'] = password
+    # add change if user exists
+    if user_change and force:
+        add_to_dict(addressx, 'FIRSTNAME', 'X')
+        add_to_dict(addressx, 'LASTNAME', 'X')
+        add_to_dict(addressx, 'E_MAIL', 'X')
+        # define Password
+        add_to_dict(passwordx, 'BAPIPWD', 'X')
+        # define LogonData
+        add_to_dict(logondatax, 'USTYP', 'X')
+        # define company
+        add_to_dict(companyx, 'COMPANY', 'X')
+        #
+        params['LOGONDATAX'] = logondatax
+        params['ADDRESSX'] = addressx
+        params['COMPANYX'] = companyx
+        params['PASSWORDX'] = passwordx
+    # return dict
+    return params
+
+
+# profiles & roles must a LIST
+def user_role_assignment_build_rfc_params(roles, username):
+    rfc_table = []
+
+    if not roles:
+        return None
+
+    for role_name in roles:
+        table_row = {'AGR_NAME': role_name}
+
+        add_to_dict(table_row, 'FROM_DAT', datetime.date.today())
+        add_to_dict(table_row, 'TO_DAT', '20991231')
+
+        rfc_table.append(table_row)
+
+    return {
+        'USERNAME': username,
+        'ACTIVITYGROUPS': rfc_table
+    }
+
+
+def user_profile_assignment_build_rfc_params(profiles, username):
+    rfc_table = []
+
+    if not profiles:
+        return None
+
+    for profile_name in profiles:
+        table_row = {'BAPIPROF': profile_name}
+        rfc_table.append(table_row)
+
+    return {
+        'USERNAME': username,
+        'PROFILES': rfc_table
+    }
+
+
+def check_user(user_detail):
+    # MESSAGE return 'User XXXX does not exist'
+    if user_detail['RETURN']:
+        for sub in user_detail['RETURN']:
+            if sub['NUMBER'] == '124':
+                return False
+    return True
+
+
+def return_analysis(raw):
+    change = False
+    failed = False
+    for state in raw['RETURN']:
+        if state['TYPE'] == "E":
+            if state['NUMBER'] == '224' or state['NUMBER'] == '124':
+                change = False
+            else:
+                failed = True
```

Comment:
I guess, there is no point in completing the for-loop, once the task has failed? so we could consider aborting here.